### PR TITLE
chore: make metatadata pkg public

### PR DESCRIPTION
**Pull request by frezbo**
Make metadata pkg public.

### CODE REVIEW

**Review by smira — APPROVED** [review]
it imports a bit of internal pkgs, might be nice to add something like

```
type Arch = artifacts.Arch
```

to prevent it being internal type

File: pkg/client/client.go [hunk 1]
```diff
@@ -57,6 +57,11 @@ func New(baseURL string, options ...Option) (*Client, error) {
 	return c, nil
 }
 
+// BaseURL returns the base URL of the client.
+func (c *Client) BaseURL() string {
```
Comment (frezbo) [inline pkg/client/client.go:61]: this is to do less hacks in TF provider

**Comment by frezbo** [comment]
> it imports a bit of internal pkgs, might be nice to add something like
> 
> ```
> type Arch = artifacts.Arch
> ```
> 
> to prevent it being internal type

good now?

**Review by smira — APPROVED** [review]

**Pull request merged**